.cicd/platforms/pinned/macos-10.14-pinned.sh

```diff
@@ -3,6 +3,8 @@ set -eo pipefail
 VERSION=1
 brew update
 brew install git cmake python libtool libusb graphviz automake wget gmp pkgconfig doxygen openssl@1.1 jq || :
+curl -LO  https://raw.githubusercontent.com/Homebrew/homebrew-core/106b4b8a421dda33c79a4018c3c3816234076331/Formula/libpqxx.rb
```

Comment:
We've done stuff like this before and were burned by it. Why are we doing these sort of hacks instead of just using it as a submodule like it appears to be designed for? @larryk85 why are you against this being a submodule?

Reply:
I'm not against submodules as a whole. I'm against this particular dependency being a submodule for a few reasons.

We need clearly defined rules as to what is or is not a submodule in our codebase.  It is clear with some of the other recent additions as submodules that without solid rules this can lead to less than ideal situations (we have to install components or the entirety of the dependencies with our own installation, possibility of building certain things inefficiently for certain targets, etc).  

This is also in effect to aid in consistency with our dependency management and overall project management.  There are no questions as to what to do when or why, if we modify it or own it then it is a submodule.  Other than that, it lives outside of that space and we have to treat it like any other external dependency.  Also if that means that we need to have custom brew/repo mirrors for our needed versions to fill in certain targets, then that is a sufficient answer as well.  Once again we don't modify these, as such the need to update them is going to be low.

We don't own libpqxx and we do not modify libpqxx, therefore it makes little sense at this point to have it in our codebase because it is probably the easier thing to do.  I fully understand your points and if in the future we need a modified variant of the lib, then it can live in our codebase and install 'along-side' our main code install.

Reply:
We don't own rocksdb, we don't own yubihsm, we don't own rapidjson, and we don't own amqp-cpp either. Nor did we modify those. Not submoduling looks contrarian not consistent.

It's not just the "easier thing to do", being a submodule has many obvious engineering benefits to us and usability benefits to users:
* Can universally, consistently pin at a known good version; something this PR does multiple ways contrary to your "aid in consistency with our dependency management" goal
* Seamlessly works for pinned builds in all environments; something this PR doesn't handle in all cases AFAIK
* Easier to install the required license file; something this PR doesn't handle
* Doesn't require users to chase down a dependency that isn't provided by common distros (we should _not_ start leaning on the build scripts to do that -- those need to go away. If we need a build script we've made it too hard. Good OSS should not require some obfuscated build script)

Instead we're spending significant engineering time crafting fragile unique solutions to these issues scattered throughout the CI and build scripts in an inconsistent way.